ecosystem/indexer-grpc/indexer-grpc-table-info/src/internal_indexer_db_service.rs

```diff
@@ -15,17 +16,16 @@ pub struct InternalIndexerDBService {
 }
 
 impl InternalIndexerDBService {
-    pub fn new(db_reader: Arc<dyn DbReader>, node_config: &NodeConfig) -> Self {
-        let db_path = node_config
-            .storage
-            .get_dir_paths()
-            .default_root_path()
-            .join(INTERNAL_INDEXER_DB);
-        let rocksdb_config = node_config.storage.rocksdb_configs.index_db_config;
-        let db = Arc::new(
-            open_internal_indexer_db(db_path, &rocksdb_config)
-                .expect("Failed to open up indexer db initially"),
-        );
+    pub fn new(
+        db_reader: Arc<dyn DbReader>,
+        node_config: &NodeConfig,
+        internal_indexer_db: Option<Arc<DB>>,
+    ) -> Self {
+        let db = if let Some(db) = internal_indexer_db {
+            db
+        } else {
+            Self::get_indexer_db(node_config).expect("Failed to get internal indexer db")
+        };
```

Comment:
always pass the option in, as we chatted.

Reply:
always pass the db now. have verification at caller side